test/matrix/matrix.cpp

```diff
@@ -99,12 +112,43 @@ struct DenseWithCustomStride : DenseWithDefaultStride {
     {
         ASSERT_EQ(mtx->get_stride(), mtx->get_size()[0] + 10);
     }
+
+    static void assert_empty_state(const matrix_type* mtx)
+    {
+        ASSERT_FALSE(mtx->get_size());
+        ASSERT_EQ(mtx->get_stride(), 0);
+        ASSERT_EQ(mtx->get_num_stored_elements(), 0);
+        ASSERT_EQ(mtx->get_const_values(), nullptr);
+    }
 };
 
-struct Coo : SimpleMatrixTest<gko::matrix::Coo<matrix_value_type, int>> {};
+struct Coo : SimpleMatrixTest<gko::matrix::Coo<matrix_value_type, int>> {
+    static void assert_empty_state(const matrix_type* mtx)
+    {
+        ASSERT_FALSE(mtx->get_size());
+        ASSERT_EQ(mtx->get_num_stored_elements(), 0);
+        ASSERT_EQ(mtx->get_const_row_idxs(), nullptr);
+        ASSERT_EQ(mtx->get_const_col_idxs(), nullptr);
+        ASSERT_EQ(mtx->get_const_values(), nullptr);
+    }
+};
 
 struct CsrWithDefaultStrategy
-    : SimpleMatrixTest<gko::matrix::Csr<matrix_value_type, int>> {};
+    : SimpleMatrixTest<gko::matrix::Csr<matrix_value_type, int>> {
+    static void assert_empty_state(const matrix_type* mtx)
+    {
+        ASSERT_FALSE(mtx->get_size());
+        ASSERT_EQ(mtx->get_num_stored_elements(), 0);
+        ASSERT_NE(mtx->get_const_row_ptrs(), nullptr);
+        ASSERT_EQ(
+            mtx->get_executor()->copy_val_to_host(mtx->get_const_row_ptrs()),
+            0);
+        ASSERT_EQ(mtx->get_const_col_idxs(), nullptr);
+        ASSERT_EQ(mtx->get_const_values(), nullptr);
```

Comment:
perhaps you could extract that into a separate function which will be called by all the different CSR variants

Reply:
should `srow` also be checked here?

Reply:
I don't think it needs to be checked, it's more of an internal property.